### PR TITLE
Replace c10::invoke_result with std::invoke_result

### PR DESCRIPTION
**Pull request by cyyever**
cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @rohan-varma

### CODE REVIEW

**Review by swolchok — APPROVED** [review]
yep, c10::invoke_result is just std::invoke_result now that we require c++17!

File: torch/csrc/distributed/c10d/FileStore.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ typename c10::invoke_result_t<F> syscall(F fn) {
     }
     return rv;
   }
+  return typename std::invoke_result_t<F>{-1};
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/distributed/c10d/FileStore.cpp:79]: -1 is int, but sometimes it will want to return a long.

**Pull request closed without merging**